spec/requests/api/v2/providers/courses_spec.rb

```diff
@@ -368,6 +372,24 @@ def perform_request
     end
   end
 
+  describe 'PATCH course' do
+    let(:path) do
+      "/api/v2/providers/#{provider.provider_code}" +
+        "/courses/#{course.course_code}"
+    end
+    let(:course) { create(:course, provider: provider, site_statuses: [site_status]) }
+    let(:site_status) { build(:site_status, :new) }
+
+    before do
+      course
+    end
+
+    it 'can update the accrediting provider' do
+      patch path, headers: { 'HTTP_AUTHORIZATION' => credentials }, params: { 'course' => { 'accrediting_provider_code' => provider.provider_code } }
+      expect(JSON.parse(response.body)['data']['attributes']['accrediting_provider_code']).to eq(provider.provider_code)
```

Comment:
it looks a bit wrong

i would of expected that `provider.provider_code` is someone else provider code

`course` has the accrediting_provider

is it that you are wanting to change?

```ruby
  let(:accrediting_provider) { create(:provider) } # ie are you trying to change that?
  let(:course) { create(:course, accrediting_provider: accrediting_provider) }
  let(:provider) { create :provider, accrediting_provider: 'Y', courses: [course] }
```

```
on course model
#  accrediting_provider_id   :integer
```

fyi this should cause/effect the expected out come that you would want

https://github.com/DFE-Digital/manage-courses-backend/blob/e3b773c26f94b6bfb6212e7515f7f130a7727dbc/app/models/course.rb#L182